src/pages/integrations/bitpay-card/bitpay-card-home/bitpay-card-home.ts

```diff
@@ -16,18 +20,29 @@ import { PhaseOneCardIntro } from '../bitpay-card-phases/phase-one/phase-one-int
 export class BitPayCardHome implements OnInit {
   public appName: string;
   public firstViewCardPhases: string;
+  public disableAddCard: boolean;
   @Input() showBitpayCardGetStarted: boolean;
   @Input() public bitpayCardItems: any;
   @Input() cardExperimentEnabled: boolean;
 
   constructor(
     private appProvider: AppProvider,
     private navCtrl: NavController,
-    private iabCardProvider: IABCardProvider
+    private iabCardProvider: IABCardProvider,
+    private events: Events,
+    private persistenceProvider: PersistenceProvider
   ) {}
 
   ngOnInit() {
     this.appName = this.appProvider.info.userVisibleName;
+    this.events.subscribe('reachedCardLimit', () => {
```

Comment:
I'm not sure this event was triggered. The `publish` action could be triggered before the subscriber (since it's in a provider). Would you mind to explain the purpose of the `events` here?

Also, any events should be unsubscribed to avoid duplicated subscribers.

Reply:
So the publish comes from card.ts and cards.ts when cards are added. This component is a singleton its on init is only ever called once. I added the event to update it in case you disconnected your bitpay id then re connected it. This way it has a default state from the persistence provider and any changes that happen after it has a subscribe thats ready to update it if changes occur. The alternative was to create an observable off of the card items but this seemed easier.

Reply:
Agreed. I prefer an event than observable too. Thanks!